### PR TITLE
feat: train v2 yaml and dto

### PR DESCRIPTION
**Pull request by jsolaas**
## Why is this pull request needed?

Start of a train v2, yaml and dto

## What does this pull request change?

- Add YamlTrain and dto.Train
- Rename Stream to StreamConditions as we have both Stream (from -> to) and StreamConditions (rate, pressure)

### CODE REVIEW

File: src/libecalc/dto/base.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class ComponentType(str, Enum):
     GENERIC = "GENERIC"
     # TURBINE = "TURBINE"
     DIRECT_EMITTER = "DIRECT_EMITTER"
+    TRAIN_V2 = "TRAIN_V2"
```
Comment (olelod) [inline src/libecalc/dto/base.py:30]: TRAIN_V2 = "TRAIN@v2" to be consistent with the rest?

File: src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py [hunk 2]
```diff
@@ -11,87 +11,33 @@
...
-                    consumer.energy_usage_model, target_period=target_period
-                ).items()
-            },
-        )
-    else:
-        raise ValueError("Unknown consumer type")
-
-
-TYamlConsumer = TypeVar("TYamlConsumer", bound=Union[YamlCompressor, YamlPump])
+TYamlConsumer = TypeVar("TYamlConsumer", bound=Union[YamlCompressor, YamlPump, YamlTrain[YamlCompressor]])
```
Comment (olelod) [inline src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py:40]: In theory you can also make a YamlTrain[YamlPump]? Even though I guess we already agreed that we have no pump trains...

File: src/tests/libecalc/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json [hunk 3]
```diff
@@ -1472,6 +1563,51 @@
...
+                                "TRAIN_V2"
+                            ],
+                            "title": "Type",
+                            "type": "string"
+                        }
+                    },
+                    "required": [
+                        "NAME",
+                        "STAGES"
+                    ],
```
Comment (olelod) [inline src/tests/libecalc/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json:1608]: Should STREAMS be required as well? Not much happening without "pipes" connecting the stages somehow...?

File: src/libecalc/presentation/yaml/yaml_types/yaml_stream_conditions.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        description="Pressure..",
+    )
+    temperature: Optional[YamlTemperature] = Field(
+        None,
+        title="Temperature",
+        description="Temperature...",
+    )
+    fluid_density: Optional[YamlDensity] = Field(None, title="Fluid density", description="The fluid density...")
+    # fluid model
+    # At least one should be specified, rate or pressure? temperature, fluid model when multistream? fluid density for pumps?
```
Comment (olelod) [inline src/libecalc/presentation/yaml/yaml_types/yaml_stream_conditions.py:64]: For compressor (trains) you will always need the fluid model (neqsim calculations) to calculate the enthalpy change/work performed over the compressor. Any stream entering a compressor network from the outside should have a fluid model, I guess.

File: src/tests/libecalc/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json (hunk 3 above)
Comment (jsolaas) [reply src/tests/libecalc/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json:1608]: I initially implemented default streams, so not required, but I removed it/postponed it. So we'll see, I guess

File: src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py (hunk 2 above)
Comment (jsolaas) [reply src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py:40]: Yes

File: src/libecalc/dto/components.py [hunk 5]
```diff
@@ -176,9 +176,29 @@ class PumpComponent(BaseConsumer):
     energy_usage_model: Dict[datetime, PumpModel]
 
 
+class Stream(EcalcBaseModel):
```
Comment (TeeeJay) [inline src/libecalc/dto/components.py:179]: Imo this is a `pipe` ... between different `equipment`...

File: src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py [hunk 6]
```diff
@@ -130,8 +74,7 @@ def to_dto(
         fuel: Optional[Dict[datetime, dto.types.FuelType]] = None,
     ) -> dto.components.ConsumerSystem:
         consumers = [
-            map_consumer(
-                consumer,
+            consumer.to_dto(
```
Comment (TeeeJay) [inline src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py:77]: nice, more fluent :)

**Review by TeeeJay — APPROVED** [review]
Exciting stuff :) lets discuss the naming :)

**Pull request merged**